web_src/css/repo.css

```diff
@@ -1313,8 +1313,9 @@
   padding: 5px 10px;
 }
 
-.repository #commits-table thead th:first-of-type {
-  padding-left: 15px;
+.repository #commits-table {
+  display: block;
```

Comment:
Hmm, should we really alter default `display: table`?

Reply:
I found [this](https://developer.mozilla.org/en-US/docs/Web/CSS/display#tables) also about it:

> In some browsers, changing the display value of a [<table>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) element to block, grid, or flex will alter its representation in the [accessibility tree](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility#accessibility_apis). This will cause the table to no longer be announced properly by screen reading technology.

Reply:
It makes the table scrollable again, but in a proper way
(that's how it looked before the original fix:)
![Screenshot_20230705-202251.png](https://github.com/go-gitea/gitea/assets/47871822/9a4ed02e-0e4f-462d-a957-9b032bf7855d)

Reply:
Do you have any better idea?

Reply:
To me this change seems only worse. with default display, table lines wrap neatly:

<img width="781" alt="Screenshot 2023-07-05 at 20 25 15" src="https://github.com/go-gitea/gitea/assets/115237/4a20da10-a8d4-45e2-acea-4a4acfe9e35f">

With this change table lines no longer wrap, break out of their container element:

<img width="778" alt="Screenshot 2023-07-05 at 20 25 06" src="https://github.com/go-gitea/gitea/assets/115237/7385ee57-fab1-4197-9391-2d9344b45850">

I would much prefer to not alter table `display`, and I think these `white-space: nowrap` are dangerous and should likely be removed. Tables are pretty good at squishing already without any additional CSS changes.

Reply:
Remove `fixed` again and keep default `display` value seems to work if keeping `white-space: nowrap`....

Reply:
Basically we need for Date, SHA and Author to stay in one line and for Message to wrap. That is being achieved by setting global `white-space: nowrap` and `white-space: normal` for message. If we then remove `fixed` class, the table also is able to scroll if needed, without altering `display`